Source/Core/DolphinQt/Debugger/CodeWidget.cpp

```diff
@@ -51,7 +51,13 @@ CodeWidget::CodeWidget(QWidget* parent) : QDockWidget(parent)
   connect(&Settings::Instance(), &Settings::DebugModeToggled,
           [this](bool enabled) { setHidden(!enabled || !Settings::Instance().IsCodeVisible()); });
 
-  connect(&Settings::Instance(), &Settings::EmulationStateChanged, this, &CodeWidget::Update);
+  connect(&Settings::Instance(), &Settings::EmulationStateChanged, [this]() {
+    if (Core::GetState() == Core::State::Paused)
```

Comment:
This can be converted into a guard clause.

```suggestion
    if (Core::GetState() != Core::State::Paused)
      return;
```